src/D2L.Security.OAuth2/Keys/Development/InMemoryPublicKeyDataProvider.cs

```diff
@@ -12,9 +12,9 @@ namespace D2L.Security.OAuth2.Keys.Development {
 	/// </summary>
 	[Obsolete( "Only use this in tests and for prototyping without a db" )]
 	public sealed class InMemoryPublicKeyDataProvider : IPublicKeyDataProvider {
-		private readonly ConcurrentDictionary<Guid, JsonWebKey> m_keys = new ConcurrentDictionary<Guid, JsonWebKey>();
+		private readonly ConcurrentDictionary<string, JsonWebKey> m_keys = new ConcurrentDictionary<string, JsonWebKey>();

```

Comment:
I know this is only the InMenory one, so this feedback _might_ apply elsewhere. This dictionary should probably get a comparer specified...

For Guids-as-strings, I want to say `OrdinalIgnoreCase`, but that won't work in general. So, perhaps just `Ordinal`? We don't want "equivalent" characters comparing as equal.

Reply:
Maybe we can do the `Guid -> string` first? That's mostly what this PR is, but it would be cool to not have to think about the new JwksProvider at the exact same time?

Reply:
Sure, I can build a new PR just for the `Guid -> string` conversion. Any other preferred changes in addition to the `Ordinal` suggestion?

Reply:
Don't have an immediate reference, but the public key caching, selecting a key from `JsonWebKeySet` and the "key to verify with" selection will need a look at for similar impacts.